src/ORM/Connect/MySQLiConnector.php

```diff
@@ -121,6 +121,13 @@ public function connect($parameters, $selectDB = false)
         if (!empty($collation)) {
             $this->dbConn->query("SET collation_connection = {$collation}");
         }
+
+        // We need to do this otherwise the client character set doesn't get applied correctly since we may be
+        // using a different collation to the default.
+        // @see http://php.net/manual/en/mysqli.set-charset.php#121647
+        if (!empty($collation) && !empty($charset)) {
+            $this->dbConn->query("SET NAMES {$charset} COLLATE {$collation}");
```

Comment:
This is called out as not recommended in the PHP docs

> This is the preferred way to change the charset. Using mysqli_query() to set it (such as SET NAMES utf8) is not recommended. See the MySQL character set concepts section for more information.

See 
http://nz2.php.net/manual/en/mysqlinfo.concepts.charset.php

Perhaps the fix here is to check if set_charset returns false and if so, throw an error?

Reply:
For whatever reason, `SET collation_connection` above isn't working. It would be good to understand why. Calling set_charset followed by SET NAMEs is probably no worse than calling set_charset followed by SET collation_connection, as long as the same charset is given to both. It's a little redundant, though,.

Still — the relevant thing that SET NAMES does is in turn call SET collation_connection, so there's something fishy going on.